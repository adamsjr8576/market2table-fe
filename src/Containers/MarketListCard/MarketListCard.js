import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './MarketListCard.scss';
// import { addVendors } from '../../actions';
import images from '../../images/images';
import { createMarketVendorLink } from '../../apiCalls';

export const MarketListCard = ({ vendorId, name, id }) => {

  const handleAddSubmit = () => {
    const vendorIdInt = parseInt(vendorId);
    const marketIdInt = parseInt(id);
    createMarketVendorLink(marketIdInt, vendorIdInt)
      .then(data => console.log(data))
      .catch(error => console.log(error))
  }

  return (
    <article className='market-list-card-article'>
      <h3 className='market-list-card-header'>{name}</h3>
      <button className='market-list-card-add-button' type='button' onClick={handleAddSubmit}>Add</button>
    </article>
  )
}

export default MarketListCard;
