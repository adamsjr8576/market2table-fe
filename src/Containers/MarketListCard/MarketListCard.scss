.market-list-card-article {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-around;
  border-bottom: 2px solid black;
  padding: 10px;
}

.market-list-card-header {
  width: 75%;
  text-align: left;
  font-size: 1.5em;
  margin: 0;
}

.market-list-card-add-button {
  width: 10%;
  font-size: 1em;
  border: 2px solid black;
  border-radius: 10px;
  cursor: pointer;
  outline: 0;
}

.market-list-card-add-button:hover {
  background-color: lightgrey;
}

.market-list-card-add-button:active {
  transform: scale(.95);
}
