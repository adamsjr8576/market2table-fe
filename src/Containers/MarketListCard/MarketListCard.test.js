import React from 'react';
import { shallow } from 'enzyme';
import { MarketListCard } from './MarketListCard';

describe('MarketListCard', () => {

  let wrapper;
  let mockvendorId = '234';
  let mockName = 'Golden Farmers Market';
  let mockId = '456';
  let mockMarketsLinked = [{name: 'Golden Farmers Market', id:'456'}, {name: 'Arvada Farmers Market', id:'8372'}];
  let mockSetMarketsLinked = jest.fn();
  let mockVendorDescription = 'A farm specializing in miniture sized produce';

  it('should match the snapshot with a vendor description', () => {
    wrapper = shallow(<MarketListCard
      vendorId={mockvendorId}
      name={mockName}
      id={mockId}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
      vendorDescription={mockVendorDescription}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without a vendor description', () => {
    wrapper = shallow(<MarketListCard
      vendorId={mockvendorId}
      name={mockName}
      id={mockId}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
