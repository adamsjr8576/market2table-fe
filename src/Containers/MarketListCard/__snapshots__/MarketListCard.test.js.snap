// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketListCard should match the snapshot with a vendor description 1`] = `
<article
  className="market-list-card-article"
>
  <h3
    className="market-list-card-header"
  >
    Golden Farmers Market
  </h3>
  <button
    className="market-list-card-add-button"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</article>
`;

exports[`MarketListCard should match the snapshot without a vendor description 1`] = `
<article
  className="market-list-card-article"
>
  <h3
    className="market-list-card-header"
  >
    Golden Farmers Market
  </h3>
  <button
    className="market-list-card-remove-button"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</article>
`;
